extends CharacterBody2D


const SPEED = 700
const JUMP_VELOCITY = -2000


func _physics_process(delta: float) -> void:
	# Add the gravity.
	if not is_on_floor():
		velocity += get_gravity() * delta

	# Handle jump.
	if Input.is_action_just_pressed("jump") and is_on_floor():
		velocity.y = JUMP_VELOCITY
	# Get the input direction and handle the movement/deceleration.
	# As good practice, you should replace UI actions with custom gameplay actions.
	var run_multiplayer = 1
	if Input.is_action_pressed("run") :
		run_multiplayer = 2
	else :
		run_multiplayer = 1
	var direction := Input.get_axis("left", "right")
	if direction:
		velocity.x = direction * SPEED * run_multiplayer
	else:
		velocity.x = move_toward(velocity.x, 0, SPEED)
	if velocity.x < 0 :
		$AnimatedSprite2D.flip_h = true
	else :
		$AnimatedSprite2D.flip_h=false
	if velocity.x != 0 :
		$AnimatedSprite2D.play("walk")
	else :
		$AnimatedSprite2D.play("idle")
	move_and_slide()
	if Input.is_action_pressed("magic"):
		var magicNode = load("res://scences/magicarea.tscn")
		var newMagic = magicNode.instantiate()
		if $AnimatedSprite2D.flip_h == false :
			newMagic.direction = -1
		else :
			newMagic.direction = 1
		newMagic.set_position(%MagicSpawnPoint.global_transform.origin)
		get_parent().add_child(newMagic)
func KillPlayer() :
	position = %respawpoint.position
	$AnimatedSprite2D.flip_h= false 
func _on_deatharea_body_entered(body: Node2D) -> void:
	KillPlayer()


func _on_endpoint_body_entered(body: Node2D) -> void:
	get_tree().change_scene_to_file("res://scences/game_end.tscn")
